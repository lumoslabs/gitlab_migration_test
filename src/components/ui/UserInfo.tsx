import React from 'react'
import { Modal, Container, Row, Col } from 'react-bootstrap'
import { css, StyleSheet } from 'aphrodite/no-important'
import commonStyles from '@styles/commonStyles'
import { base } from '@styles/colors'
import WideActionButton from '@components/ui/WideActionButton'

export interface IUserInfoProps {
  show: boolean;
  name?: string;
  email?: string;
  profilePicUrl?: string;
  timezone?: string;
  handleClose(e: React.MouseEvent<any>): any;
  logoutCallback(e: React.MouseEvent<any>): any;
}

const UserInfo = ({ show, handleClose, profilePicUrl, email, name, timezone, logoutCallback }: IUserInfoProps): JSX.Element => {

  return (
    <Modal className={css(styles.modal)}
      show={show}
      onHide={handleClose}
    >
      <Modal.Title>
        <p className={css(styles.title)}>
          {'Your Profile'}
        </p>
      </Modal.Title>
      <Modal.Body>
        <div className={css([commonStyles.flexColumnAlignCenter, styles.avatarDiv])}>
          <img
            src={profilePicUrl || '/assets/guest_avatar.svg'}
            className={css(styles.avatarImg)}
            alt='Avatar'
          />
        </div>
        <Container>
          <Row>
            <Col xs={4}>
              <p className={css(styles.labelsText)}>
                {'Email'}
              </p>
            </Col>
            <Col xs={8}>
              <p className={css(styles.valueText, styles.email)}>
                {email || ''}
              </p>
            </Col>
          </Row>
          <Row>
            <Col xs={4}>
              <p className={css(styles.labelsText)}>
                {'Name'}
              </p>
            </Col>
            <Col xs={8}>
              <p className={css(styles.valueText)}>
                {name}
              </p>
            </Col>
          </Row>
          <Row>
            <Col xs={4}>
              <p className={css(styles.labelsText)}>
                {'Time Zone'}
              </p>
            </Col>
            <Col xs={8}>
              <p className={css(styles.valueText)}>
                {timezone || 'Pacific Time USA & Canada'}
              </p>
            </Col>
          </Row>
          <Row>
            <Col xs={4}>
              <p className={css(styles.labelsText)}>
                {'Language'}
              </p>
            </Col>
            <Col xs={8}>
              <p className={css(styles.valueText)}>
                {'English'}
              </p>
            </Col>
          </Row>
        </Container>
        <WideActionButton
          extendStyles={styles.cta}
          buttonText='Logout'
          onClick={logoutCallback}
          eventData={{id: 'logout', message: 'Logout' }}
        />
      </Modal.Body>
    </Modal>
  )
}

const styles = StyleSheet.create({
  modal: {
    background: base.lumosWhite,
    borderRadius: '4.66vh',
    height: '96vh',
    width: '70vw',
    top: '2vh',
    left: '15vw',
    boxShadow: `0px 1px 1px 0px rgba(0, 0, 0, 0.11),
      0px 2px 4px 0px rgba(0, 0, 0, 0.13),
      0px 2px 6px 0px rgba(0, 0, 0, 0.1)`
  },

  title: {
    color: base.lumosBlack,
    fontFamily: 'MuseoSans700',
    fontSize: '4vh',
    fontWeight: 700,
    display: 'flex',
    margin: '0px',
    padding: '0px',
    marginLeft: '3.12vw',
    marginTop: '5.33vh'
  },

  avatarDiv: {
    marginBottom: '3vh'
  },

  avatarImg: {
    width: '15vh',
    height: '15vh',
    borderRadius: '50%'
  },

  labelsText: {
    color: base.lumosBlack,
    fontFamily: 'MuseoSans700',
    fontSize: '3.33vh',
    fontWeight: 700
  },

  valueText: {
    color: base.lumosBlack,
    fontFamily: 'MuseoSans500',
    fontSize: '3.33vh',
    fontWeight: 500,
    textAlign: 'left',
    flex: 1,
    marginRight: '2.34vw'
  },

  email: {
    whiteSpace: 'nowrap',
    overflow: 'hidden',
    textOverflow: 'ellipsis'
  },

  cta: {
    height: '10vh',
    width: '15vw',
    borderRadius: '36px'
  }
})

export default UserInfo
