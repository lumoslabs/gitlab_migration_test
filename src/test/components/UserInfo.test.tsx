import React from 'react'
import UserInfo from '@components/ui/UserInfo'
import { render } from '@testing-library/react'

const handleClick = () => { window.location.href = '/' }

it('matches snapshot', () => {
  const { asFragment } = render(
    <UserInfo
      show={true}
      handleClose={handleClick}
      logoutCallback={handleClick}
      email='email@example.com'
      name='Lumosity User'
      timezone='Pacific Time'
    />, {}
  )
  expect(asFragment()).toMatchSnapshot()
})