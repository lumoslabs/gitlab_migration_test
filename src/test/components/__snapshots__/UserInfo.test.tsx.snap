// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBar displays the correct info 1`] = `
<DocumentFragment>
  <div
    class="fade modal-backdrop show"
  />
  <div
    aria-modal="true"
    class="fade modal_d8krus modal show"
    role="dialog"
    style="display: block;"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-title h4"
        >
          <p
            class="title_1809nyj"
          >
            Your Profile
          </p>
        </div>
        <div
          class="modal-body"
        >
          <div
            class="flexColumnAlignCenter_1ndpwxc-o_O-avatarDiv_1f3o2jp"
          >
            <img
              alt="Avatar"
              class="avatarImg_106i4z8"
              src="/assets/guest_avatar.svg"
            />
          </div>
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="col-4"
              >
                <p
                  class="labelsText_1hh284c"
                >
                  Email
                </p>
              </div>
              <div
                class="col-8"
              >
                <p
                  class="valueText_91azc3-o_O-email_1iurgbx"
                >
                  email@example.com
                </p>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-4"
              >
                <p
                  class="labelsText_1hh284c"
                >
                  Name
                </p>
              </div>
              <div
                class="col-8"
              >
                <p
                  class="valueText_91azc3"
                >
                  Lumosity User
                </p>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-4"
              >
                <p
                  class="labelsText_1hh284c"
                >
                  Time Zone
                </p>
              </div>
              <div
                class="col-8"
              >
                <p
                  class="valueText_91azc3"
                >
                  Pacific Time
                </p>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-4"
              >
                <p
                  class="labelsText_1hh284c"
                >
                  Language
                </p>
              </div>
              <div
                class="col-8"
              >
                <p
                  class="valueText_91azc3"
                >
                  English
                </p>
              </div>
            </div>
          </div>
          <div
            class="flexJustifyCenter_jfsfb5"
          >
            <button
              class="button_ssaaah-o_O-wideAction_4zoj1f-o_O-cta_2yfb43"
            >
              Logout
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="fade modal-backdrop show"
  />
  <div
    aria-modal="true"
    class="fade modal_d8krus modal show"
    role="dialog"
    style="display: block;"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-title h4"
        >
          <p
            class="title_1809nyj"
          >
            Your Profile
          </p>
        </div>
        <div
          class="modal-body"
        >
          <div
            class="flexColumnAlignCenter_1ndpwxc-o_O-avatarDiv_1f3o2jp"
          >
            <img
              alt="Avatar"
              class="avatarImg_106i4z8"
              src="/assets/guest_avatar.svg"
            />
          </div>
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="col-4"
              >
                <p
                  class="labelsText_1hh284c"
                >
                  Email
                </p>
              </div>
              <div
                class="col-8"
              >
                <p
                  class="valueText_91azc3-o_O-email_1iurgbx"
                >
                  email@example.com
                </p>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-4"
              >
                <p
                  class="labelsText_1hh284c"
                >
                  Name
                </p>
              </div>
              <div
                class="col-8"
              >
                <p
                  class="valueText_91azc3"
                >
                  Lumosity User
                </p>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-4"
              >
                <p
                  class="labelsText_1hh284c"
                >
                  Time Zone
                </p>
              </div>
              <div
                class="col-8"
              >
                <p
                  class="valueText_91azc3"
                >
                  Pacific Time
                </p>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-4"
              >
                <p
                  class="labelsText_1hh284c"
                >
                  Language
                </p>
              </div>
              <div
                class="col-8"
              >
                <p
                  class="valueText_91azc3"
                >
                  English
                </p>
              </div>
            </div>
          </div>
          <div
            class="flexJustifyCenter_jfsfb5"
          >
            <button
              class="button_ssaaah-o_O-wideAction_4zoj1f-o_O-cta_2yfb43"
            >
              Logout
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
